### DISCUSSION TITLE
Unable to download artifact(s): Unable to download and extract artifact: Request timeout

### DISCUSSION THREAD

**Discussion by jijiechen** [community/community#81413] [Actions] [answered]
### Select Topic Area

Product Feedback

### Body

We updated to v4 of the download-artifact action plugin recently and we are seeing request timeouts after that updating.

In the run below, this error had caused two job failures while other jobs in the same workflow and run succeeded, please take a look:

[Unable to download artifact(s): Unable to download and extract artifact: Request timeout](https://github.com/kumahq/kuma/actions/runs/7291088227/job/19869652348#step:9:21)

```
Downloading single artifact
Preparing to download the following artifacts:
- build-output (ID: 1129435572, Size: 838[18](https://github.com/kumahq/kuma/actions/runs/7291088227/job/19869652348#step:9:19)7798)
Redirecting to blob download url: https://productionresultssa3.blob.core.windows.net/actions-results/57bb6424-dc03-4623-b418-78822b7c3d7d/workflow-job-run-ca395085-040a-526b-2ce8-bdc85f692774/artifacts/build-output.zip
Starting download of artifact to: /home/runner/work/kuma/kuma/build
Error: Unable to download artifact(s): Unable to download and extract artifact: Request timeout: /actions-results/57bb6424-dc03-4623-b418-78822b7c3d7d/workflow-job-run-ca395085-040a-526b-2ce8-bdc85f692774/artifacts/build-output.zip?se=[20](https://github.com/kumahq/kuma/actions/runs/7291088227/job/19869652348#step:9:21)23-12-[21](https://github.com/kumahq/kuma/actions/runs/7291088227/job/19869652348#step:9:22)T17%3A26%3A03Z&sig=tOYxonEtsVrT%2FRGs6tO1vto64FzOeJ94s7ZA14BknKA%3D&sp=r&spr=https&sr=b&st=2023-12-21T17%3A16%3A03Z&sv=2021-12-02
```

People are also discussing this issue at code repo: https://github.com/actions/download-artifact/issues/249

**Comment by fzyzcjy** [comment]
Hi, I am seeing the same, is there any solutions!


https://github.com/fzyzcjy/flutter_rust_bridge/actions/runs/7296903359/job/19887619400?pr=1472

```
Starting download of artifact to: /home/runner/work/flutter_rust_bridge/flutter_rust_bridge/downloaded-artifacts/test_dart_native--ubuntu-20.04--frb_dart--coverage
Artifact download completed successfully.
Artifact download completed successfully.
Artifact download completed successfully.
Artifact download completed successfully.
/home/runner/work/_actions/actions/download-artifact/v4/dist/index.js:2120
            throw new Error(`Unable to download and extract artifact: ${error.message}`);
                  ^

Error: Unable to download and extract artifact: Request timeout: /actions-results/83f822ab-6457-42d0-b7c8-47c80637e43a/workflow-job-run-5d323ea9-244b-5416-b3ac-6ad640de49e8/artifacts/test_rust--windows-2019----coverage.zip?se=2023-12-22T08%3A35%3A12Z&sig=OjkR619JF2rsJWej5GUT8y0ndUKK34bSMswrpNJSmRQ%3D&sp=r&spr=https&sr=b&st=2023-12-22T08%3A25%3A12Z&sv=2021-12-02
    at /home/runner/work/_actions/actions/download-artifact/v4/dist/index.js:2120:19
    at Generator.throw (<anonymous>)
    at rejected (/home/runner/work/_actions/actions/download-artifact/v4/dist/index.js:1999:65)

**Reply by zaikunzhang** [reply]
The same here: 
```
Error: Unable to download and extract artifact: Request timeout: /actions-results/b3cbc986-d648-4f87-a362-ddb060a05cf4/workflow-job-run-c506f3c0-2e80-593c-e758-49f715b8e661/artifacts/cobylal-small-classical-yes-2.zip?se=2023-12-26T23%3A19%3A43Z&sig=F7U5gLg0kuzSqcCJVQxOLwWCRP6jHaXrh87L3WSd4IQ%3D&sp=r&spr=https&sr=b&st=2023-12-26T23%3A09%3A43Z&sv=2021-12-02
    at /home/runner/work/_actions/actions/download-artifact/v4.0.0/dist/index.js:2120:19
    at Generator.throw (<anonymous>)
    at rejected (/home/runner/work/_actions/actions/download-artifact/v4.0.0/dist/index.js:1999:65)
    at runNextTicks (node:internal/process/task_queues:60:5)
    at listOnTimeout (node:internal/timers:540:9)
    at process.processTimers (node:internal/timers:514:7)
```

**Reply by felixfontein** [reply]
I'm also seeing this a lot in https://github.com/getsops/sops/ over the last few days. Restarting the failed job usually fixes the problem, but it's really annoying.

Edit: just after sending this, https://github.com/getsops/sops/actions/runs/7340220059/job/19985952777?pr=1392 needed *two* retries to finally succeed.

**Comment by bitomaxsp** [comment]
Same here:
```
Unable to download artifact(s): Unable to download and extract artifact: Request timeout: /actions-results/5a34ff42-aea4-429e-8556-e9efa3390bb4/workflow-job-run-d3b3501f-fe8a-51ea-4fe1-048236f14daa/artifacts/matrixfiles.zip?se=2023-12-29T08%3A16%3A42Z&sig=xQTWXGn%2FlBZ4QiQJMObQgnF%2BwX0bZXEWcGKW9hKUA2g%3D&sp=r&spr=https&sr=b&st=2023-12-29T08%3A06%3A42Z&sv=2021-12-02
```

**Comment by lyubomir-trifonov** [comment]
[**EDIT - I found my issue**]
I was running jobs in parallel, and I was actually trying to download the artifact from second job, BEFORE the first job that creates it was finished.

A simple `needs: [xxxxx]` ordering fixed the issue for me.

---- original post below ---
Same here:

Run 1 - 5 seconds after the artifact is reported as created:
```
Run actions/download-artifact@v4
Downloading single artifact
Error: Unable to download artifact(s): Artifact not found for name: xxxxx-v45
```

Re-run after 1 minute:
```
Downloading single artifact
Preparing to download the following artifacts:
- xxxxxxxxxxxxxx-v45 (ID: 11yyyy83531, Size: 214067396)
Downloading artifact 'xxxxxxxx'
Redirecting to blob download url: https://productionresultssa4.blob.core.windows.net/actions-results/xxxxxxxx/artifacts/yyyyyyy-v45.zip
Starting download of artifact to: /home/runner/work/xxxxxxxxci-demo/package
Error: Unable to download artifact(s): Unable to download and extract artifact: Request timeout: /actions-results/c67fb27c-5776-489e-a58b-027c55ec9fa6/xxxxxxxxx-v45.zip?se=2023-12-[29](https://xxxxxxxxxxxx#step:5:30)T12%3A06%3A[30](https://xxxxxxxxxx383#step:5:31)Z&sig=KdcxKStha%2yyyyyyyy8lEVIRcxEv25NQk%3D&sp=r&spr=https&sr=b&st=2023-12-29T11%3A56%3A30Z&sv=2021-12-02
```

Third attempt in 5 minutes succeeds.

then, 4th attempt fails again...

Seems like the artifact is behind a distributed LB storage, where the content is not yet synced to all backend nodes to me.

**Comment by cooper667** [comment]
Also seeing this a LOT after moving to v4 of the action

**Comment by webknjaz** [comment]
Sounds like something I reported years ago: https://github.com/actions/toolkit/issues/1235. It was with older artifact APIs but the underlying problem appears to be the same.

**Reply by jijiechen** [reply]
Thanks for the information, but we did not update the artifact after its uploading.

**Comment by nmattia** [comment]
Seeing a lot of this (on every which includes ~20 jobs that download artifacts) after upgrading to v4. To be clear, we do wait for the producing job to complete with `needs: <job that uploads artifact>`.

@robherley any idea what might be happening here?

[link to run](https://github.com/dfinity/internet-identity/actions/runs/7448151714/job/20262132667?pr=2179#step:9:14)

**Comment by edumserrano** [comment]
I'm also experiencing the same timeout issue.

> Run actions/download-artifact@v4
>
> Downloading single artifact
>
> Preparing to download the following artifacts:
> -repo (ID: 1156559649, Size: 156565819)
>
> Redirecting to blob download url: https://productionresultssa3.blob.core.windows.net/actions-results/2048dadb-6887-4923-8141-47e21eae4f6e/workflow-job-run-d85a69e5-6c6b-5883-b18d-364516809cdf/artifacts/repo.zip
>
> Starting download of artifact to: /home/runner/work/playwright-adventures/playwright-adventures
>
> Error: Unable to download artifact(s): Unable to download and extract artifact: Request timeout: /actions-results/2048dadb-6887-4923-8141-47e21eae4f6e/workflow-job-run-d85a69e5-6c6b-5883-b18d-364516809cdf/artifacts/repo.zip?se=2024-01-09T11%3A51%3A38Z&sig=iXD29va1Ow0bmT33f9R%2Fb9Bqh1Er3hga%2FaO9oQPOrSw%3D&sp=r&spr=https&sr=b&st=2024-01-09T11%3A41%3A38Z&sv=2021-12-02

[link to run](https://github.com/edumserrano/playwright-adventures/actions/runs/7460635680/job/20299111764?pr=47)

**Comment by UlyssesZh** [comment]
[Also seeing this.](https://github.com/UlyssesZh/UlyssesZh/actions)

**Comment by nmattia** [comment]
This seems to be much more stable/reliable now! I'm able to run 60+ artifact upload/downloads without a single failure

**Comment by andrewakim** [comment, marked as answer]
Thanks for sharing this feedback! We pushed out [a fix last week](https://github.com/actions/download-artifact/pull/274) to address the timeout issues folks were experiencing with the artifact actions. We're also actively addressing it on the backend as well. If you're still experiencing this issue regularly, can you please open up a [new issue in actions/download-artifact](https://github.com/actions/download-artifact/issues)? Thanks again!
